source/validate.cpp

```diff
@@ -261,6 +267,143 @@ spv_result_t spvValidateIDs(const spv_instruction_t* pInsts,
   return SPV_SUCCESS;
 }
 
+// TODO(umar): Validate header
+spv_result_t setHeader(void* user_data, spv_endianness_t endian, uint32_t magic,
+                       uint32_t version, uint32_t generator, uint32_t id_bound,
+                       uint32_t reserved) {
+  (void)user_data;
+  (void)endian;
+  (void)version;
+  (void)generator;
+  (void)id_bound;
+  (void)reserved;
+  return SPV_SUCCESS;
+}
+
+// TODO(umar): Move this class to another file
+class validation_state_t {
+  const uint32_t* binary;
+  spv_diagnostic* diagnostic;
+  const uint32_t* program_counter;
+
+  std::set<uint32_t> declared_ids;
+  std::set<uint32_t> forward_ids;
+
+ public:
+  validation_state_t(const uint32_t* binary, spv_diagnostic* diag)
+      : binary(binary), diagnostic(diag), program_counter(binary) {}
+
+  void set_instruction_words(const uint32_t * const words) {
```

Comment:
Much as I dislike camelcase (and I really do), convention is to use camelcase for method names.
Style guide says method names start with a captial.  The rest of the SPIR-V Tools hasn't caught up to that yet.